plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/local-pickup-select/index.tsx

```diff
@@ -14,7 +14,11 @@ interface LocalPickupSelectProps {
 	pickupLocations: CartShippingPackageShippingRate[];
 	renderPickupLocation: (
 		location: CartShippingPackageShippingRate,
-		pickupLocationsCount: number
+		pickupLocationsCount: number,
+		// This is the ID of the rate that is selected in the _client_ (not necessarily the selected shipping rate on the server, yet)
+		// If the server returns a cart with a different selected shipping rate, then after "receiving" the updated cart (`receiveCart`)
+		// this arg, `clientSelectedOption`, will change to be the ID for that rate, and the UI will update.
+		clientSelectedOption?: string
```

Comment:
I'm thinking we can skip the `client` from the name here.

`selectedOption` should be enough; it's usually clear what it means, whether it comes from the client or the server.